src/Microsoft.VisualStudio.Threading/AwaitExtensions.cs

```diff
@@ -160,7 +160,9 @@ private static async Task WaitForRegistryChangeAsync(SafeRegistryHandle registry
                         // subscription to have begun before we return: for the async part to simply be notification.
                         // This async method we're calling uses .ConfigureAwait(false) internally so this won't
                         // deadlock if we're called on a thread with a single-thread SynchronizationContext.
+#pragma warning disable CA2000 // Dispose objects before losing scope
                         dedicatedThreadReleaser = DownlevelRegistryWatcherSupport.ExecuteOnDedicatedThreadAsync(registerAction).GetAwaiter().GetResult();
+#pragma warning restore CA2000 // Dispose objects before losing scope
```

Comment:
📝 I believe this one is a false positive in the analyzer. I'm going to file a follow-up issue on dotnet/roslyn-analyzers for it.